### PR TITLE
Add component information client and server

### PR DESCRIPTION
**Pull request by julianoes**
Depends on https://github.com/mavlink/MAVSDK-Proto/pull/270.

Probably best reviewed commit by commit.

### CODE REVIEW

**Review by JonasVautherin — DISMISSED** [review]
Nice!

**Review by JonasVautherin — APPROVED** [review]

**Review by JonasVautherin — APPROVED** [review]

File: src/mavsdk/core/mavlink_request_message_handler.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    _system_impl(system_impl)
+{
+    _system_impl.register_mavlink_command_handler(
+        MAV_CMD_REQUEST_MESSAGE,
+        [this](const MavlinkCommandReceiver::CommandLong& command) {
+            return handle_command_long(command);
+        },
+        this);
+
+    _system_impl.register_mavlink_command_handler(
```
Comment (dayjaby) [inline src/mavsdk/core/mavlink_request_message_handler.cpp:16]: Is the duplicate register_mavlink_command_handler on purpose?

File: src/mavsdk/core/mavlink_request_message_handler.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class SystemImpl;
+
+class MavlinkRequestMessageHandler {
+public:
+    MavlinkRequestMessageHandler() = delete;
+    explicit MavlinkRequestMessageHandler(SystemImpl& system_impl);
+    ~MavlinkRequestMessageHandler();
+
+    using Params = std::array<float, 5>;
+    using Callback = std::function<std::optional<MAV_RESULT>(Params)>;
```
Comment (dayjaby) [inline src/mavsdk/core/mavlink_request_message_handler.h:22]: Consider `const Params&` instead of `Params` here.

**Review by dayjaby — COMMENTED** [review]
Looking good so far!

File: src/mavsdk/core/mavlink_request_message_handler.cpp (hunk 1 above)
Comment (julianoes) [reply src/mavsdk/core/mavlink_request_message_handler.cpp:16]: Ooops, good catch! I think one should be for command_int!

File: src/mavsdk/core/mavlink_request_message_handler.h (hunk 2 above)
Comment (julianoes) [reply src/mavsdk/core/mavlink_request_message_handler.h:22]: Yes, would make sense. Thx.

**Review by JonasVautherin — APPROVED** [review]

File: src/mavsdk/core/mavlink_request_message_handler.h (hunk 2 above)
Comment (julianoes) [reply src/mavsdk/core/mavlink_request_message_handler.h]: Changed in 71ff6ae5, finally.

**Comment by julianoes** [comment]
I think this is somewhat contained and should not be interfering with the rest too much, so I'm merging it as in its current prototype state.

**Pull request merged**